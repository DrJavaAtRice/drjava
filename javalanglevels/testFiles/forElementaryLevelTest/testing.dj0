//this makes sure that autoboxing is correctly done in the case where the object type corresponding to the
//primitive is not in the symbol table.

abstract class AList {
}

class Empty extends AList {
}

class NonEmpty extends AList {
  Object first;
  AList rest;
}

class UseLists {
  int myMethod() { 
    AList a = new NonEmpty(2, new Empty());
    return 5;
  }
}