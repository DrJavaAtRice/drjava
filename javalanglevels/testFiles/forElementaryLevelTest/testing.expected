// Language Level Converter line number map: dj*->java. Entries: 20
//     1->5         2->6         3->7         4->8         5->9         6->40        7->41        8->42   
//     9->73       10->74       11->75       12->76       13->77       14->125      15->126      16->127  
//    17->128      18->129      19->130      20->131  
//this makes sure that autoboxing is correctly done in the case where the object type corresponding to the
//primitive is not in the symbol table.

abstract class AList {

  /** This method is automatically generated by the Language Level Converter. */
  public AList() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      AList cast = ((AList) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}

class Empty extends AList {

  /** This method is automatically generated by the Language Level Converter. */
  public Empty() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      Empty cast = ((Empty) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}

class NonEmpty extends AList {
  private final Object first;
  private final AList rest;

  /** This method is automatically generated by the Language Level Converter. */
  public NonEmpty(java.lang.Object first, AList rest) {
    super();
    this.first = first;
    this.rest = rest;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.Object first() {
    return first;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public AList rest() {
    return rest;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        first() + ", " + 
        rest() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      NonEmpty cast = ((NonEmpty) o);
      return (first() != null && first().equals(cast.first())) && 
          (rest() != null && rest().equals(cast.rest()));
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        (first() == null ? 0 : first().hashCode()) ^ 
        (rest() == null ? 0 : rest().hashCode());
  }
}

class UseLists {
  public int myMethod() { 
    final AList a = new NonEmpty(2, new Empty());
    return 5;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public UseLists() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      UseLists cast = ((UseLists) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
